src/jaguar.toit

```diff
@@ -88,35 +91,43 @@ run-installed-containers -> none:
 
 serve arguments:
   device := Device.parse arguments
-  while true:
-    attempts ::= 3
-    failures := 0
-    while failures < attempts:
-      exception := catch: run device
-      // If we have a pending firmware upgrade, we take care of
-      // it before trying to re-open the network.
-      if firmware-is-upgrade-pending: firmware.upgrade
-      // If Jaguar needs to be disabled, we stop here and wait until
-      // we can re-enable Jaguar.
-      if disabled:
-        network-free.up      // Signal to start running the container.
-        container-done.down  // Wait until done running the container.
-        disabled = false
-        continue
-      // Log exceptions and count the failures so we can back off
-      // and avoid excessive attempts to re-open the network.
-      if exception:
-        failures++
-        logger.warn "running Jaguar failed due to '$exception' ($failures/$attempts)"
-    // If we need to validate the firmware and we've failed to do so
-    // in the first round of attempts, we roll back to the previous
-    // firmware right away.
-    if firmware-is-validation-pending:
-      logger.error "firmware update was rejected after failing to connect or validate"
-      firmware.rollback
-    backoff := Duration --s=5
-    logger.info "backing off for $backoff"
-    sleep backoff
+  endpoints := [
```

Comment:
It would be nice to get these endpoints passed in to `serve` from the outside, so the configuration step is nicely split from the logic.

Reply:
will address this in a follow-up PR.